sdk/textanalytics/Azure.AI.TextAnalytics/src/RecognizeCustomEntitiesResultCollection.cs

```diff
@@ -0,0 +1,88 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using System.Linq;
+
+namespace Azure.AI.TextAnalytics
+{
+    /// <summary>
+    /// Collection of <see cref="RecognizeEntitiesResult"/> objects corresponding
+    /// to a batch of documents, and information about the batch operation.
+    /// </summary>
+    [DebuggerTypeProxy(typeof(RecognizeCustomEntitiesResultCollectionDebugView))]
+    public class RecognizeCustomEntitiesResultCollection : ReadOnlyCollection<RecognizeEntitiesResult>
+    {
+        internal RecognizeCustomEntitiesResultCollection(IList<RecognizeEntitiesResult> list, TextDocumentBatchStatistics statistics, string projectName, string deploymentName) : base(list)
+        {
+            Statistics = statistics;
+            DeploymentName = deploymentName;
+            ProjectName = projectName;
+        }
+
+        /// <summary>
+        /// Gets statistics about the documents and how it was processed
+        /// by the service.  This property will have a value when IncludeStatistics
+        /// is set to true in the client call.
+        /// </summary>
+        public TextDocumentBatchStatistics Statistics { get; }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the deployment.
+        /// </summary>
+        public string DeploymentName { get; }
+
+        /// <summary>
+        /// Gets the value of the property corresponding to the name of the project.
+        /// </summary>
+        public string ProjectName { get; }
+
+        /// <summary>
+        /// Debugger Proxy class for <see cref="RecognizeCustomEntitiesResultCollection"/>.
+        /// </summary>
+        internal class RecognizeCustomEntitiesResultCollectionDebugView
+        {
+            private RecognizeCustomEntitiesResultCollection BaseCollection { get; }
+
+            public RecognizeCustomEntitiesResultCollectionDebugView(RecognizeCustomEntitiesResultCollection collection)
+            {
+                BaseCollection = collection;
+            }
+
+            [DebuggerBrowsable(DebuggerBrowsableState.RootHidden)]
+            public List<RecognizeEntitiesResult> Items
```

Comment:
did you consider duplicating `RecognizeEntitiesResult` class to have a name with custom in it? /cc @maririos

Reply:
not really as the service has told us that they will maintain the same structure as non custom recognized entitites

Reply:
You're right both will have the same structure. I was just wondering if it is worth it to use a name with custom in it to not confuse the customer.

Reply:
Same as in Java. Reuse the existing model in NRE

Reply:
Ohh I see. Not really. I think as is it can also tell the customer that they will get the same information structure as if they use the prebuilt functions.
Because of this is inside a Custom class, I think it will be ok.

We can always see if UX Studies or feedback from users suggest the opposite and do the rename.

Reply:
https://apiview.dev/Assemblies/Review/f62ee261035a44189c597868c7a8a7ee?diffRevisionId=6a15977bd7974ddca84777f7975941ae&doc=False&diffOnly=True#com.azure.ai.textanalytics.util.RecognizeCustomEntitiesResultCollection